src/driver/driver_api.cc

```diff
@@ -128,63 +128,192 @@ transform::Pass Filter(FCond fcond) {
   return tir::transform::CreatePrimFuncPass(fpass, 0, "Filter", {});
 }
 
-IRModule lower(te::Schedule sch, const Array<te::Tensor>& args, const std::string& name,
-               const std::unordered_map<te::Tensor, tir::Buffer>& binds) {
-  Array<ObjectRef> out_arg_list;
+Array<tvm::transform::Pass> CreatePassList(bool simple_mode, bool legacy_te_pass) {
```

Comment:
Clarification : what is legacy_te_pass mean here?

Reply:
Legacy te pass means that the object we are lowering is a te::Schedule. I've copied the logic and phrasing directly from the python version -- I don't know the details of why this is still here. But without it, some tests fail.

Reply:
Maybe we could be more specific in the name ? e.g. using such as "bool from_te_schedule" ?

Also I see simple_mode is just enabling LoopPartition, we could use "bool enable_loop_partition" ?